multimodal/src/autogluon/multimodal/models/mmdet_image.py

```diff
@@ -293,45 +298,39 @@ def image_feature_dim(self):
 
     def forward(
         self,
-        batch: dict,
+        batch,
+        mode,
     ):
         """
         Parameters
         ----------
         batch
             A dictionary containing the input mini-batch data.
             We need to use the keys with the model prefix to index required data.
+        mode
+            "loss" or "predict". TODO: support "tensor"
+            https://github.com/open-mmlab/mmdetection/blob/main/mmdet/models/detectors/base.py#L58C1
 
         Returns
         -------
             A dictionary with bounding boxes.
         """
-        # TODO: refactor this to work like forward() in MMDet, and support realtime predict
-        logger.warning("MMDetAutoModelForObjectDetection.forward() is deprecated since it does not support multi gpu.")
 
-        data = batch[self.image_key]
-
-        data["img_metas"] = [img_metas.data[0] for img_metas in data["img_metas"]]
-        data["img"] = [img.data[0] for img in data["img"]]
-
-        device = next(self.model.parameters()).device  # model device
-        if next(self.model.parameters()).is_cuda:
-            # scatter to specified GPU
-            data = scatter(data, [device])[0]
+        self.set_data_preprocessor_device()
+        data = self.data_preprocessor(batch)
+        rets = self.model(
+            inputs=data["inputs"],
+            data_samples=data["data_samples"],
+            mode=mode,
+        )
+
+        if mode == "loss":
+            return rets
+        elif mode == "predict":
+            # for detailed data structure, see https://github.com/open-mmlab/mmdetection/blob/main/mmdet/structures/det_data_sample.py
+            return [{BBOX: ret.pred_instances, LABEL: ret.gt_instances} for ret in rets]
```

Comment:
Why does `predict` mode require groundtruth labels?

Reply:
`model(mode="predict")` is used for both predicting and evaluation. Here `gt_instances` will be None if there is no label available. It's easier to include the ground truth here than merging ground truth with result in lit module / predictor.